common/lib/xmodule/xmodule/contentstore/content.py

```diff
@@ -330,17 +334,28 @@ def generate_thumbnail(self, content, tempfile_path=None, dimensions=None):
         pixels. It defaults to None.
         """
         thumbnail_content = None
+        is_svg = content.content_type == 'image/svg+xml'
         # use a naming convention to associate originals with the thumbnail
         thumbnail_name = StaticContent.generate_thumbnail_name(
-            content.location.name, dimensions=dimensions
+            content.location.name, dimensions=dimensions, extension='.svg' if is_svg else None
         )
         thumbnail_file_location = StaticContent.compute_location(
             content.location.course_key, thumbnail_name, is_thumbnail=True
         )
 
         # if we're uploading an image, then let's generate a thumbnail so that we can
         # serve it up when needed without having to rescale on the fly
-        if content.content_type is not None and content.content_type.split('/')[0] == 'image':
+        if is_svg:
+            # for svg simply store the provided svg file, since vector graphics should be good enough
+            # for downscaling client-side
+            if tempfile_path is None:
+                thumbnail_file = StringIO.StringIO(content.data)
+            else:
+                thumbnail_file = open(tempfile_path)
```

Comment:
It is my understanding that if you open a file, you need to close it. You can do this with try/finally and error handling, or you can use the nicer "with" syntax. http://stackoverflow.com/questions/8774830/how-with-is-better-than-try-catch-to-open-a-file-in-python